### PR TITLE
Add nutanix support with basic lifecycle testcases

### PR DESCRIPTION
**Pull request by shi2wei3**
Signed-off-by: Wei Shi <wshi@redhat.com>

### CODE REVIEW

File: config/nutanix.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Cloud:
```
Comment (litian1992) [inline config/nutanix.yaml:1]: Need also to add config/test_nutanix.yaml which should refer to nutanix.yaml, nutanix_flavors.yaml and nutanix_testcases.yaml.

File: tests/nutanix/test_functional_lifecycle.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            return
+        if self.name.name.endswith("test_create_vm_sshkey"):
+            pre_delete = True
+        if self.name.name.endswith("test_start_vm"):
+            pre_stop = True
+        self.session = self.cloud.init_vm(pre_delete=pre_delete,
+                                          pre_stop=pre_stop)
+
+    def test_create_vm_password(self):
+        self.vm.create(wait=True)
```
[the remark this answers is not in the document]
Comment (shi2wei3) [reply tests/nutanix/test_functional_lifecycle.py:26]: Split keypair and passwd assignment in user-data now, and pubkey str will be fetched from local machine.
Please see my new commit.

File: config/nutanix.yaml (hunk 1 above)
Comment (shi2wei3) [reply config/nutanix.yaml:1]: Thanks for the remind, added as described.

**Review by litian1992 — APPROVED** [review]
All looks good. Managed to create VM on Nutanix, start, ssh then delete.

**Review by yuxisun1217 — APPROVED** [review]
LGTM.

**Pull request merged**